enhancements/installer/feature-sets.md

```diff
@@ -0,0 +1,209 @@
+)
+```
+
+`ExperimentalNoUpgrade`, for Installer work in progress or
+experimental features, is added to the set of accepted feature sets.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Install Config
+
+The basic implementation through the install config is relatively straight forward. Gated features are protected
+by validation:
+
+```go
+
+	if c.FeatureSet == nil || *c.FeatureSet != types.TechPreviewNoUpgrade {
+		errMsg := "the TechPreviewNoUpgrade feature set must be enabled to use this field"
+
+    if installConfig.Foo != nil && len(installConfig.Foo.Bar) > 0 {
+			allErrs = append(allErrs, field.Forbidden(field.NewPath("foo", "bar"), errMsg))
+		}
+	}
+```
+
+Package `types` defines the install config and represents the API offered by the Installer. Most notably this
+API is consumed/used by hive. For this reason, it is important that feature sets be part of the install config.
+
+#### CLI Flag (future work)
+
+On the other hand, the install config is an asset and gated features could potentially be exercised
+before the install config is validated (or perhaps the feature is completely independent from the install config). For this reason, it would
+be a good idea to provide a flag:
+
+```shell
+$ openshift-install create cluster --feature-set TechPreviewNoUpgrade
+```
+
+This flag, if provided, would also populate the value in the install config; and conflicts between the flag and install config
+would throw an error.
+
+While there are clear use cases for a flag, there are no immediate use cases and the implementation of the Installer (with
+its separation between command-line flags and assets) does not lead to a clear-cut design. Implementing this functionality
+is certainly possible, but I propose we defer implementation for this until it is determined to be necessary.
+
+#### openshift-install explain
+
+`explain` provides a command-line reference for the install config:
+
+```shell
+$ openshift-install explain installconfig | tail -n 6
+    pullSecret <string> -required-
+      PullSecret is the secret to use when pulling images.
+
+    sshKey <string>
+      SSHKey is the public Secure Shell (SSH) key to provide access to instances.
+```
+
+`explain` should handle feature-gated fields appropriately, either by skipping them (not printing them) or clearly marking
+them as gated features.
+
+### Risks and Mitigations
+
+I am not aware of any significant risks to introducing this feature. 
+
+### Drawbacks
+
+I am not aware of any significant drawbacks to introducing this feature.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. The [Tech Preview Guidelines](https://github.com/openshift/enhancements/blob/master/guidelines/techpreview.md)
+state that Tech Preview features, and perhaps all gated features, need to be listed in the [OpenShift API](https://github.com/openshift/api/blob/bace76a807222b30bb9bfd4926826348156fb522/config/v1/types_feature.go#L117). Is that necessary for the Installer, or
```

Comment:
I expect the installer to list their individual features as well so that we can track graduation.